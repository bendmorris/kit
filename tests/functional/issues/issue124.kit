#[promote] abstract MyType: CArray[Int, 4] {
    rules {
        ($this[$a] = $b) => printf("set: %s = %s\n", $a, $b);
        ($this[$a]) => printf("get: %s\n", $a);

        ($this.abc = $x) => printf("field write: %s\n", $x);
        ($this.abc) => printf("field read\n");
    }
}

function main() {
    var a: MyType = [1, 2, 3, 4];
    a["test1"] = "test2";
    a["test3"];

    a.abc = "test4";
    a.abc;
}
