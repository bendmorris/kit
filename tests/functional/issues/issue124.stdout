set: test1 = test2
get: test3
field write: test4
field read
